File: Recursive/hanoi.py
def move(N, start, to):
    print(f"Move {N} disk, from {start} to {to}")

def hanoi(N, start, via, to):
    if N == 1:
        move(1, start, to)
    else:
        hanoi(N-1, start, to, via)
        move(N, start, to)
        hanoi(N-1, via, start, to)
        
if __name__ == '__main__':
    n = int(input())
    hanoi(n, 'A', 'B', 'C')

File: Recursive/hanoi.c
#include <stdio.h>

void move(int n, char start, char end){
    printf("move %d disk, from %c to %c \n", n, start, end);
};

void hanoi(int n, char start, char via, char end){
    if(n == 1){
            move(n, start, end);}
    else{
        hanoi(n-1, start, end, via);
        move(n, start ,end);
        hanoi(n-1, start, via, end);}
};

void main(){

    int N;
    scanf("%d", &N);
    hanoi(N, 'a', 'b', 'c');
    return 0;

};


